pkg/controller/deployment_inplace.go

```diff
@@ -0,0 +1,559 @@
+			err := dc.annotateNodesBackingMachineSets(ctx, allISs, clusterAutoscalerScaleDownAnnotations)
+			if err != nil {
+				klog.Errorf("Failed to add %s on all nodes. Error: %s", clusterAutoscalerScaleDownAnnotations, err)
+				return err
+			}
+		}
+	}
+
+	if err != nil {
+		klog.Warningf("Failed to add %s on all nodes. Error: %s", PreferNoScheduleKey, err)
+	}
+
+	// label all the machines and nodes backing the old machine sets as candidate for update
+	err = dc.labelNodesBackingMachineSets(ctx, oldISs, v1alpha1.LabelKeyNodeCandidateForUpdate, "true")
+	if err != nil {
+		return fmt.Errorf("failed to label nodes backing old machine sets as candidate for update: %v", err)
+	}
+
+	if err := dc.syncMachineSets(ctx, oldISs, newIS, d); err != nil {
+		return err
+	}
+
+	// In this section, we will attempt to scale up the new machine set. Machines with the `node.machine.sapcloud.io/update-successful` label
+	// can transfer their ownership to the new machine set.
+	// It is crucial to ensure that during the ownership transfer, the machine is not deleted,
+	// and the old machine set is not scaled up to recreate the machine.
+	scaledUp, err := dc.reconcileNewMachineSetInPlace(ctx, oldISs, newIS, d)
+	if err != nil {
+		klog.V(3).Infof("this was unexpected error")
+		return err
+	}
+	if scaledUp {
+		// Update DeploymentStatus
+		return dc.syncRolloutStatus(ctx, allISs, newIS, d)
+	}
+
+	// prepare old ISs for update
+	machinesSelectedForUpdate, err := dc.reconcileOldMachineSetsInPlace(ctx, allISs, FilterActiveMachineSets(oldISs), newIS, d)
+	if err != nil {
+		return err
+	}
+	if machinesSelectedForUpdate {
+		// Update DeploymentStatus
+		return dc.syncRolloutStatus(ctx, allISs, newIS, d)
+	}
+
+	if err := dc.syncMachineSets(ctx, oldISs, newIS, d); err != nil {
+		return err
+	}
+
+	if MachineDeploymentComplete(d, &d.Status) {
+		if dc.autoscalerScaleDownAnnotationDuringRollout {
+			// Check if any of the machine under this MachineDeployment contains the by-mcm annotation, and
+			// remove the original autoscaler annotation only after.
+			err := dc.removeAutoscalerAnnotationsIfRequired(ctx, allISs, clusterAutoscalerScaleDownAnnotations)
+			if err != nil {
+				return err
+			}
+		}
+		if err := dc.cleanupMachineDeployment(ctx, oldISs, d); err != nil {
+			return err
+		}
+	}
+
+	// Sync deployment status
+	return dc.syncRolloutStatus(ctx, allISs, newIS, d)
+}
+
+func (dc *controller) syncMachineSets(ctx context.Context, oldISs []*v1alpha1.MachineSet, newIS *v1alpha1.MachineSet, deployment *v1alpha1.MachineDeployment) error {
+	machines, err := dc.machineLister.List(labels.SelectorFromSet(newIS.Spec.Selector.MatchLabels))
+	if err != nil {
+		return err
+	}
+
+	machinesWithUpdateSuccessfulLabel := filterMachinesWithUpdateSuccessfulLabel(machines)
+	klog.V(3).Infof("machine with update successful label in new machine set %v", len(machinesWithUpdateSuccessfulLabel))
```

Comment:
If your intention is to log number of machines then you should use `%d`
Else if you wanted to log machine names, then `%v` is fine

Also, please supply the exact label name with `%q`